rust/arrow/test/data/nested_structs.json

```diff
@@ -0,0 +1,4 @@
+{"a": {"b": true, "c": {"d": "text"}}}
```

Comment:
Given the size of this file (very small) I wonder if the test case itself would be more readable if the contents were inlined directly into the rust source code. I don't see any issues with it being in its own file, this is more a style comment

Reply:
I opened https://issues.apache.org/jira/browse/ARROW-10764